tests/phpunit/integration/Modules/Analytics_4Test.php

```diff
@@ -1666,6 +1667,262 @@ public function test_get_conversion_events( $access_token ) {
 		$this->assertEquals( "/v1beta/properties/$property_id/conversionEvents", $request_url['path'] );
 	}
 
+	public function test_get_enhanced_mesurement_settings__required_params() {
+		// Grant READONLY_SCOPE so request doesn't fail.
+		$this->authentication->get_oauth_client()->set_granted_scopes(
+			array_merge(
+				$this->authentication->get_oauth_client()->get_required_scopes(),
+				(array) Analytics::READONLY_SCOPE
+			)
+		);
+
+		$data = $this->analytics->get_data(
+			'enhanced-measurement-settings',
+			array()
+		);
+
+		// Verify if the propertyID is required.
+		$this->assertWPErrorWithMessage( 'Request parameter is empty: propertyID.', $data );
+		$this->assertEquals( 'missing_required_param', $data->get_error_code() );
+		$this->assertEquals( array( 'status' => 400 ), $data->get_error_data( 'missing_required_param' ) );
+
+		$data = $this->analytics->get_data(
+			'enhanced-measurement-settings',
+			array(
+				'propertyID' => '123456789',
+			)
+		);
+
+		// Verify if the webDataStreamID is required.
+		$this->assertWPErrorWithMessage( 'Request parameter is empty: webDataStreamID.', $data );
+		$this->assertEquals( 'missing_required_param', $data->get_error_code() );
+		$this->assertEquals( array( 'status' => 400 ), $data->get_error_data( 'missing_required_param' ) );
+	}
+
+	public function test_get_enhanced_mesurement_settings() {
+		$property_id        = '123456789';
+		$web_data_stream_id = '654321';
+
+		$this->analytics->get_settings()->merge(
+			array(
+				'propertyID'      => $property_id,
+				'webDataStreamID' => $web_data_stream_id,
+			)
+		);
+
+		// Grant READONLY_SCOPE so request doesn't fail.
+		$this->authentication->get_oauth_client()->set_granted_scopes(
+			array_merge(
+				$this->authentication->get_oauth_client()->get_required_scopes(),
+				(array) Analytics::READONLY_SCOPE
+			)
+		);
+
+		FakeHttp::fake_google_http_handler(
+			$this->analytics->get_client(),
+			$this->create_fake_http_handler( $property_id )
+		);
+		$this->analytics->register();
+
+		// Fetch enhanced measurement settings.
+		$data = $this->analytics->get_data(
+			'enhanced-measurement-settings',
+			array(
+				'propertyID'      => $property_id,
+				'webDataStreamID' => $web_data_stream_id,
+			)
+		);
+
+		$this->assertNotWPError( $data );
+
+		$data_array = (array) $data;
+
+		// Assert that the keys exist.
+		$keys = array(
+			'fileDownloadsEnabled',
+			'name',
+			'outboundClicksEnabled',
+			'pageChangesEnabled',
+			'scrollsEnabled',
+			'searchQueryParameter',
+			'siteSearchEnabled',
+			'streamEnabled',
+			'uriQueryParameter',
+			'videoEngagementEnabled',
+		);
+
+		foreach ( $keys as $key ) {
+			$this->assertArrayHasKey( $key, $data_array );
+		}
+
```

Comment:
As discussed above, with a mock object being returned by the fake HTTP handler, we can add an additional assertion here. E.g., assuming the mock object is returning `'streamEnabled' => true`, we can do as follows. This should also be used in other tests as appropriate (I think that would just be the `test_set_enhanced_measurement_settings()` case).

```suggestion

		// Verify the enhanced measurement settings are returned by checking a field value.
		$this->assertEquals( true, $data['streamEnabled'] );

```

Reply:
Updated ✅